titus-server-master/src/main/java/com/netflix/titus/master/scheduler/resourcecache/OpportunisticCpuAllocation.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2019 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.titus.master.scheduler.resourcecache;
+
+import java.util.Objects;
+
+public class OpportunisticCpuAllocation {
+    private final String taskId;
+    private final String agentId;
+    private final String allocationId;
+    private final int cpuCount;
+
+    public OpportunisticCpuAllocation(String taskId, String agentId, String allocationId, int cpuCount) {
+        this.taskId = taskId;
+        this.agentId = agentId;
+        this.allocationId = allocationId;
+        this.cpuCount = cpuCount;
+    }
+
+    public String getTaskId() {
+        return taskId;
+    }
+
+    public String getAgentId() {
+        return agentId;
+    }
+
+    public String getAllocationId() {
+        return allocationId;
+    }
+
+    public int getCpuCount() {
+        return cpuCount;
+    }
+
+    @Override
+    public String toString() {
+        return "OpportunisticCpuAllocation{" +
+                "taskId='" + taskId + '\'' +
+                ", agentId='" + agentId + '\'' +
+                ", allocationId='" + allocationId + '\'' +
+                ", cpuCount=" + cpuCount +
+                '}';
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (!(o instanceof OpportunisticCpuAllocation)) {
+            return false;
+        }
+        OpportunisticCpuAllocation that = (OpportunisticCpuAllocation) o;
+        return cpuCount == that.cpuCount &&
+                taskId.equals(that.taskId) &&
+                agentId.equals(that.agentId) &&
+                allocationId.equals(that.allocationId);
+    }
+
+    public static OpportunisticCpuAllocationBuilder newBuilder() {
+        return new OpportunisticCpuAllocationBuilder();
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(taskId, agentId, allocationId, cpuCount);
+    }
+
+    public static final class OpportunisticCpuAllocationBuilder {
```

Comment:
Just `Builder` to align with the rest of the code?

Reply:
will do, I used a plugin to autogenerate this